android/app/src/main/java/org/fossasia/openevent/activities/SettingsActivity.java

```diff
@@ -73,8 +74,10 @@ public boolean onPreferenceChange(Preference preference, Object o) {
         } else if (preference.getKey().equals(getResources().getString(R.string.timezone_mode_key))) {
             if (o.equals(false)) {
                 timezonePreference.setChecked(false);
+                ISO8601Date.setTimeZone();
             } else if (o.equals(true)) {
                 timezonePreference.setChecked(true);
+                ISO8601Date.setTimeZoneDefault();
```

Comment:
Yeah @arp95 I think although this is not your change but it can be fixed to be just `else` and also rename the `o` variable while you are at it. Sorry for so many requirement changes